### ISSUE TITLE
Already disposed: Project (Disposed)

### ISSUE THREAD

**Issue by aristotll** [zielu/GitToolBox#42]
```log
Already disposed: Project (Disposed) snow-parent
java.lang.AssertionError: Already disposed: Project (Disposed) snow-parent
	at com.intellij.openapi.components.impl.ComponentManagerImpl.getComponent(ComponentManagerImpl.java:167)
	at zielu.gittoolbox.GitToolBoxProject.getInstance(GitToolBoxProject.java:24)
	at zielu.gittoolbox.fetch.AutoFetchTask.doFetch(AutoFetchTask.java:109)
	at zielu.gittoolbox.fetch.AutoFetchTask.doFetch(AutoFetchTask.java:88)
	at zielu.gittoolbox.fetch.AutoFetchTask.access$000(AutoFetchTask.java:25)
	at zielu.gittoolbox.fetch.AutoFetchTask$1.run(AutoFetchTask.java:125)
	at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:726)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:176)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:556)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:501)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:66)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:163)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl$1.run(ProgressManagerImpl.java:137)
	at com.intellij.openapi.application.impl.ApplicationImpl$2.run(ApplicationImpl.java:334)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

The project snow has been closed before, but then the exception occurs.

**Comment by zielu** [comment]
Another one in version with fix

```log
java.lang.AssertionError: Already disposed: Project (Disposed) <redacted>
	at com.intellij.openapi.components.impl.ComponentManagerImpl.getPicoContainer(ComponentManagerImpl.java:257)
	at com.intellij.openapi.components.impl.ComponentManagerImpl.getPicoContainer(ComponentManagerImpl.java:56)
	at com.intellij.project.ProjectKt.getStateStore(project.kt:31)
	at com.intellij.openapi.project.impl.ProjectImpl.getStateStore(ProjectImpl.java:196)
	at com.intellij.openapi.project.impl.ProjectImpl.getBaseDir(ProjectImpl.java:229)
	at com.intellij.vcsUtil.VcsImplUtil.getShortVcsRootName(VcsImplUtil.java:49)
	at com.intellij.dvcs.DvcsUtil.getShortRepositoryName(DvcsUtil.java:118)
	at git4idea.repo.GitRepositoryImpl.readRepoInfo(GitRepositoryImpl.java:207)
	at git4idea.repo.GitRepositoryImpl.update(GitRepositoryImpl.java:201)
	at git4idea.update.GitFetcher.fetch(GitFetcher.java:83)
	at zielu.gittoolbox.fetch.GtFetcher.fetchRoots(GtFetcher.java:66)
	at zielu.gittoolbox.fetch.AutoFetchTask.doFetch(AutoFetchTask.java:108)
	at zielu.gittoolbox.fetch.AutoFetchTask.doFetch(AutoFetchTask.java:88)
	at zielu.gittoolbox.fetch.AutoFetchTask.access$100(AutoFetchTask.java:25)
	at zielu.gittoolbox.fetch.AutoFetchTask$1.run(AutoFetchTask.java:125)
	at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:718)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:170)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:548)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:493)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:94)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:157)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl$2.run(ProgressManagerImpl.java:165)
	at com.intellij.openapi.application.impl.ApplicationImpl$2.run(ApplicationImpl.java:343)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by zielu** [comment]
Fixed in 171.1.4
